proto/upgrades/astria/upgrades/v1/upgrades.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package astria.upgrades.v1;
+
+import "astria/protocol/genesis/v1/types.proto";
+
+// A JSON-encoded form of this message is used as the upgrades file for the Sequencer.
+message Upgrades {
+  Upgrade1 upgrade_1 = 1;
+}
+
+// Info specific to a given upgrade.
+//
+// All upgrades have this info at a minimum.
+message BaseUpgradeInfo {
+  // The upgrade should be applied during the lifecycle of the block at this height.
+  uint64 activation_height = 1;
+  // The app version running after the upgrade is applied.
+  uint64 app_version = 2;
+}
+
+// Upgrade 1 of the Sequencer network.
+message Upgrade1 {
+  message ConnectOracleChange {
+    // The Connect oracle genesis data.
+    astria.protocol.genesis.v1.ConnectGenesis genesis = 1;
+  }
+
+  message ValidatorUpdateActionChange {}
+
+  BaseUpgradeInfo base_info = 1;
+  ConnectOracleChange connect_oracle_change = 2;
+  ValidatorUpdateActionChange validator_update_action_change = 3;
+}
+
+// Brief details of a given upgrade change.
+//
+// This is NOT part of the upgrades file schema, rather it is used by the sequencer gRPC service in
+// the `GetUpgradesInfo` response.
+message ChangeInfo {
```

Comment:
maybe move to `service.proto` then?

Reply:
Good point.  Done in [4a045d3](https://github.com/astriaorg/astria/pull/1917/commits/4a045d36c177676c3af883aed37da3cce61f30b6).